infra/monitoring.yaml

```diff
@@ -0,0 +1,225 @@
+    Description: The ELB to alert on.
+  LBTGName:
+    Type: String
+    Description: The ELB TargetGroup to alert on.
+
+Resources:
+  # Alerts that do not require paging - shouldn't interrupt someone or wake them up.
+  ticketalerts:
+    Type: AWS::SNS::Topic
+    Properties:
+      TopicName: UATTicketAlerts
+      DisplayName: UATTicketAlerts
+  # Alerts that require immediate action and may indicate that the site is down.
+  # We want to get someone out of bed for this.
+  pagingalerts:
+    Type: AWS::SNS::Topic
+    Properties:
+      TopicName: UATPagingAlerts
+      DisplayName: UATPagingAlerts
+
+  # Send a page if more than 5% of requests are 500-range errors.
+  # This probably means that the website is seriously broken for a large percentage
+  # of users.
+  5xxAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmDescription: Too many 5xx on the live site detected.
+      AlarmName: 5xxErrorsTooHigh
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      TreatMissingData: ignore
+      EvaluationPeriods: 5
+      # Is this too sensitive?  we'll have to see.
+      Threshold: 0.05
+      AlarmActions:
+        - !Ref pagingalerts
+      Metrics:
+        - Id: ratio
+          Expression: (total5xx / (total2xx + total3xx + total5xx))
+          ReturnData: true
+        - Id: total2xx
+          MetricStat:
+            Metric:
+              MetricName: HTTPCode_Target_2XX_Count
+              Namespace: AWS/ApplicationELB
+            Period: 60
+            Stat: Sum
+          ReturnData: false
+        - Id: total3xx
+          MetricStat:
+            Metric:
+              MetricName: HTTPCode_Target_3XX_Count
+              Namespace: AWS/ApplicationELB
+            Period: 60
+            Stat: Sum
+          ReturnData: false
+        - Id: total5xx
+          MetricStat:
+            Metric:
+              MetricName: HTTPCode_Target_5XX_Count
+              Namespace: AWS/ApplicationELB
+            Period: 60
+            Stat: Sum
+          ReturnData: false
+
+  # Send a ticket if more than 1% of requests are 400-range errors.
+  # This probably means there's a broken link somewhere on the site,
+  # either inbound or internal.
+  4xxAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmDescription: Too many 4xx on the live site - broken link?
+      AlarmName: 4xxErrorsTooHigh
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      TreatMissingData: ignore
+      EvaluationPeriods: 20
+      Threshold: 0.01
+      AlarmActions:
+        - !Ref ticketalerts
+      Metrics:
+        - Id: ratio
+          Expression: (total4xx / (total2xx + total3xx + total4xx))
+          ReturnData: true
+        - Id: total2xx
+          MetricStat:
+            Metric:
+              MetricName: HTTPCode_Target_2XX_Count
+              Namespace: AWS/ApplicationELB
+              Dimensions:
+                - Name: LoadBalancer
+                  Value: !Ref LBName
```

Comment:
**Question**: We are not referencing this in above alert. Do we need it? What is the difference?

Reply:
Oh, we *do* need it there.  I was *wondering* why that was showing up as "not enough data".  Thanks, good eye!